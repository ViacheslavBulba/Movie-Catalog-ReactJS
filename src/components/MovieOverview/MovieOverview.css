.overview-container {
    width: 100vw;
    display: flex;
    flex-direction: column;
    background-color: #232323;
}

/* .overview-container::before {
    content: '';
    position: absolute;
    left: 0;
    right: 0;
    z-index: -1;

    display: block;
    background-image: url('../../../public/netflix-background.jpg');
    background-size: cover;
    width: 100vw;
    height: 100%;

    -webkit-filter: blur(3px);
    -moz-filter: blur(3px);
    -o-filter: blur(3px);
    -ms-filter: blur(3px);
    filter: blur(3px) brightness(0.2);
} */

.logo-and-back-button-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-top: 20px;
    margin-bottom: 20px;
    margin-left: 60px;
    margin-right: 60px;
}

.movie-details-container {
    display: flex;
    flex-direction: row;
    margin-bottom: 80px;
    margin-left: 60px;
    margin-right: 60px;
}

.search-icon {
    font-size: 36px;
    color: #f65261;
}

.movie-image-overview {
    display: flex;
    height: 400px;
}

.overview-text-container {
    margin-left: 60px;
    margin-right: 60px;
    width: 100%;
}

.name-and-rating-container {
    display: flex;
    flex-direction: row;
    margin-top: 20px;
    line-height: 1;
}

.overview-title {
    font-size: 48px;
    color: #ffffff;
}

.overview-rating {
    font-size: 48px;
    color: #bdf84e;
    margin-left: 30px;
    border-radius: 10px;
    border: solid 1px;
    padding-left: 10px;
    padding-right: 10px;
    height: 55px;
}

.overview-tagline,
.overview-text {
    color: #ffffff;
}

.year-and-duration-container {
    display: flex;
    flex-direction: row;
    font-size: 28px;
    color: #f65261;
    margin-top: 5px;
    margin-bottom: 5px;
}

.overview-year {
    margin-right: 10px;
}

@media screen and (max-width: 929px) {
    .movie-image-overview {
        display: none;
    }

    .overview-text-container {
        margin-left: 0;
        margin-right: 0;
        /* width: 100%; */
    }

    .movie-details-container {
        display: flex;
        flex-direction: column;
        margin-bottom: 80px;
        margin-left: 60px;
        margin-right: 60px;
    }
}