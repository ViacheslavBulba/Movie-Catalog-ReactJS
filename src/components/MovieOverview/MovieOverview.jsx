import React from 'react';
import './MovieOverview.css';
import Moment from 'react-moment';
import PropTypes from 'prop-types';
import logo from '../../../public/netflix-logo.svg';
import noPicture from '../../../public/no-picture-available.jpg';

export default function MovieOverview(props) {
    return (
        <>
            <div className='overview-container'>
                <div className='logo-and-back-button-container'>
                    <img src={logo} alt='logo' className='logo' />
                    <i
                        className='fa fa-times search-icon'
                        onClick={props.closeOverview}
                    ></i>
                </div>
                <div className='movie-details-container'>
                    <img
                        className='movie-image-overview'
                        src={props.movie.poster_path || noPicture}
                    ></img>
                    <div className='overview-text-container'>
                        <div className='name-and-rating-container'>
                            <span className='overview-title'>
                                {props.movie.title}
                            </span>
                            <span className='overview-rating'>
                                {props.movie.vote_average}
                            </span>
                        </div>
                        <span className='overview-tagline'>
                            {props.movie.tagline}
                        </span>
                        <div className='year-and-duration-container'>
                            <Moment
                                date={props.movie.release_date}
                                parse='YYYY-MM-dd'
                                format='YYYY'
                                className='overview-year'
                            />
                            <span>{props.movie.runtime} min</span>
                        </div>
                        <span className='overview-text'>
                            {props.movie.overview}
                        </span>
                    </div>
                </div>
            </div>
        </>
    );
}

MovieOverview.propTypes = {
    movie: PropTypes.shape({
        id: PropTypes.number.isRequired,
        poster_path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        release_date: PropTypes.string.isRequired,
        genres: PropTypes.arrayOf(PropTypes.string.isRequired),
        vote_average: PropTypes.number.isRequired,
        tagline: PropTypes.string.isRequired,
        runtime: PropTypes.number.isRequired,
        overview: PropTypes.string.isRequired,
    }).isRequired,
    closeOverview: PropTypes.func.isRequired,
};
