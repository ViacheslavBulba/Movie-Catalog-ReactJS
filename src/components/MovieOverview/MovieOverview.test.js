import React from 'react';
import renderer from 'react-test-renderer';
import MovieOverview from './MovieOverview';

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

const initialState = {
    pending: false,
    movies: [],
    error: null,
    sortBy: 'release_date',
    sortOrder: 'desc',
    filerByGenres: [],
    search: '',
    searchBy: 'title',
    movieNotFoundById: false,
    movieToOverview: null,
};

const store = mockStore(initialState)

test('snapshot test for movie overview', () => {
    const component = renderer.create(
        <Provider store={store}>
            <MovieOverview />
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
