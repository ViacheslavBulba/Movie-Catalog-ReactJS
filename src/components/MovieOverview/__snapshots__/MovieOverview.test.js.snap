// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for movie overview 1`] = `
<span>
  Loading...
</span>
`;
